test/sql/function/string/test_like_any.test

```diff
@@ -0,0 +1,81 @@
+# name: test/sql/function/string/test_like_any.test
+# description: Test LIKE ANY with list of patterns
+# group: [string]
+
+statement ok
+PRAGMA enable_verification
+
+# basic LIKE ANY with constants (ARRAY literal)
+query T
+SELECT 'abc' LIKE ANY (ARRAY['a%', '%d'])
```

Comment:
It seems that this doesn't yet work with `NOT LIKE` or `NOT ILIKE`. I think it would be fine to leave that for a future PR. But perhaps it's not too much work to add here

```sql
SELECT 'abc' NOT LIKE ANY (ARRAY['a%', '%d']);
```
```
Parser Error: Unsupported comparison "!~~" for ANY/ALL subquery
```